test/unit/math/prim/mat/fun/cov_periodic_test.cpp

```diff
@@ -0,0 +1,854 @@
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/prim/mat/fun/distance.hpp>
+#include <stan/math/prim/mat/fun/cov_periodic.hpp>
+#include <gtest/gtest.h>
+
+template<typename T_x1, typename T_x2, typename T_sigma, typename T_l, typename T_p>
+std::string pull_msg(std::vector<T_x1> x1,
+                     std::vector<T_x2> x2,
+                     T_sigma sigma,
+                     T_l l,
+					 T_p p) {
+  std::string message;
+  try {
+    stan::math::cov_periodic(x1, x2, sigma, l, p);
+  } catch (std::domain_error& e) {
+    message = e.what();
+  } catch (...) {
+    message = "Threw the wrong exception";
+  }
+  return message;
+}
+
+template<typename T_x1, typename T_sigma, typename T_l, typename T_p>
+std::string pull_msg(std::vector<T_x1> x1,
+                     T_sigma sigma,
+                     T_l l,
+					 T_p p) {
+  std::string message;
+  try {
+    stan::math::cov_periodic(x1, sigma, l, p);
+  } catch (std::domain_error& e) {
+    message = e.what();
+  } catch (...) {
+    message = "Threw the wrong exception";
+  }
+  return message;
+}
+
+
+TEST(MathPrimMat, vec_double_cov_periodic1) {
+  double sigma = 0.2;
```

Comment:
Somewhere in this file I think could be a few Cholesky factorizations. The covariance matrix should be positive definite. If it's not, it's a bug.

Reply:
Added Cholesky factorizations to three places in tests, but not to everywhere.